### PR TITLE
[2201.4.x] Backport user-defined metadata and content-type header support

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
# Description
This PR backports the improvements (added to the master branch) with  https://github.com/ballerina-platform/module-ballerinax-aws.s3/pull/109 and https://github.com/ballerina-platform/module-ballerinax-aws.s3/pull/110

Related Pull Requests (remove if not relevant)
- https://github.com/ballerina-platform/module-ballerinax-aws.s3/pull/109
- https://github.com/ballerina-platform/module-ballerinax-aws.s3/pull/110

One line release note: 
- One line describing the feature/improvement/fix made by this PR 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* Ballerina Version:
* Operating System:
* Java SDK: 

# Checklist:

### Security checks
 - [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: s3/utils.bal [hunk 1]
```diff
@@ -241,6 +241,19 @@ isolated function populateCreateObjectHeaders(map<string> requestHeaders, Object
...
+    }
+}
+
+# Function to populate createObject optional user-defined metadata headers.
+#
+# + requestHeaders - Request headers map.
+# + userMetadataHeaders - Map containing user-defined metadata.
+isolated function populateUserMetadataHeaders(map<string> requestHeaders, map<string> userMetadataHeaders){
+    foreach string metadataKey in userMetadataHeaders.keys() {
+        requestHeaders["x-amz-meta-" + metadataKey.toLowerAscii()] = userMetadataHeaders[metadataKey] ?: "";
```
Comment (ayeshLK) [inline s3/utils.bal:256]:
```suggestion
        requestHeaders[string `x-amz-meta-${metadataKey.toLowerAscii()}`] = userMetadataHeaders.get(metadataKey);
```

File: s3/utils.bal [hunk 2]
```diff
@@ -241,6 +241,19 @@ isolated function populateCreateObjectHeaders(map<string> requestHeaders, Object
...
+        if (objectCreationHeaders?.contentType != ()) {
+            requestHeaders[CONTENT_TYPE] = <string>objectCreationHeaders?.contentType;
+        }
+    }
+}
+
+# Function to populate createObject optional user-defined metadata headers.
+#
+# + requestHeaders - Request headers map.
+# + userMetadataHeaders - Map containing user-defined metadata.
```
Comment (ayeshLK) [inline s3/utils.bal:253]: Since this is not a public method do we need to add documentation ?

Comment (NipunaRanasinghe) [reply s3/utils.bal:253]: @ayeshLK yeah its not mandatory, but users can optionally add it. So I don't see a reason why we should remove an already added documentation :)

**Review by ayeshLK — APPROVED** [review]
LGTM

**Pull request merged**